orsopy/fileio/base.py

```diff
@@ -33,8 +44,14 @@ def _clean(self):
         :return: Cleaned dictionary
         :rtype: dict
         """
-        return dict((k, v) for (k, v) in self.__dict__.items()
-                    if (v is not None or k not in self.orso_optionals))
+        d = {}
+        for k, v in self.__dict__.items():
```

Comment:
dataclasses.asdict does this for you - it collects all the fields that have been defined, recursively through every attribute that is also decorated as a dataclass (so no inheritance is needed)

Reply:
It wasn't catching the `orso_optionals` from each inherited level for me, hence the recursive method.